supertokens_python/recipe/emailverification/api/implementation.py

```diff
@@ -13,65 +13,127 @@
 # under the License.
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, Dict
+from typing import TYPE_CHECKING, Any, Dict, Union
 
 from supertokens_python.recipe.emailverification.interfaces import (
-    APIInterface, EmailVerifyPostInvalidTokenErrorResponse,
-    EmailVerifyPostOkResponse,
+    APIInterface, CreateEmailVerificationTokenOkResult,
+    EmailVerifyPostInvalidTokenErrorResponse, EmailVerifyPostOkResponse,
     GenerateEmailVerifyTokenPostEmailAlreadyVerifiedErrorResponse,
-    GenerateEmailVerifyTokenPostOkResponse, IsEmailVerifiedGetOkResponse)
+    GenerateEmailVerifyTokenPostOkResponse, IsEmailVerifiedGetOkResponse,
+    VerifyEmailUsingTokenOkResult)
+from supertokens_python.recipe.session.interfaces import SessionContainer
 
 if TYPE_CHECKING:
     from supertokens_python.recipe.emailverification.interfaces import (
-        APIOptions, GenerateEmailVerifyTokenPostResponse, IsEmailVerifiedGetResponse, EmailVerifyPostResponse
+        APIOptions, GenerateEmailVerifyTokenPostResponse
     )
 
 from supertokens_python.recipe.emailverification.types import User
 from supertokens_python.recipe.session.asyncio import get_session
 
 
 class APIImplementation(APIInterface):
-    async def email_verify_post(self, token: str, api_options: APIOptions, user_context: Dict[str, Any]) -> EmailVerifyPostResponse:
+    async def email_verify_post(self, token: str, api_options: APIOptions, user_context: Dict[str, Any]) -> Union[EmailVerifyPostOkResponse, EmailVerifyPostInvalidTokenErrorResponse]:
+        """email_verify_post is an async mehtod that accepts
+
+               - token: string,
+               - api_options: APIOptions,
+               - user_context: python dict
+
+            if the user is successfully varified, the response could be as follows
+
+                {
+                    "status": "OK",
+                    "user": {
+                      "id": "<usern-id>",
+                      "email": "<email-id>"
+                }
```

Comment:
The function doesn't return a JSON. So this is technically incorrect.

Reply:
fixed it.